sqlmesh/core/plan/evaluator.py

```diff
@@ -101,7 +101,7 @@ def evaluate(self, plan: Plan) -> None:
             self._push(plan, deployability_index)
             self._restate(plan)
             self._backfill(plan, before_promote_snapshots, deployability_index)
-            promotion_result = self._promote(plan, deployability_index)
+            promotion_result = self._promote(plan, before_promote_snapshots)
```

Comment:
Was the issue that we were getting the latest snapshots and those might have gaps but we weren't promoting latest but instead just promoting what we just backfilled?

Reply:
No, I don't think so. Those were all latest snapshots. It just some snapshots can only be evaluated after promotion, but the no gap check was performed for all of them.